### PR TITLE
chore(nextjs): Apply deprecation warnings

### PR DESCRIPTION
**Pull request by dimkl**
## Description

** REVIEW IT PER COMMIT **

Warn about deprecations of the following in Clerk Nextjs:

- `withClerkMiddleware`
- `withServerSideAuth`
- `CLERK_JS_VERSION` environment variable
- `CLERK_API_KEY` environment variable
- `NEXT_PUBLIC_FRONTEND_API` environment variable

## Checklist

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [x] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [x] (If applicable) [Documentation](https://github.com/clerkinc/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [x] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [x] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

### CODE REVIEW

File: packages/nextjs/src/shared/deprecated.ts
Comment (dimkl) [inline packages/nextjs/src/shared/deprecated.ts:1]: `@clerk/shared` package cannot be used in `@clerk/nextjs` server context due to the following error:


Screenshot 2023-09-21 at 21 32 30

Comment (LekoArts) [inline packages/nextjs/src/shared/deprecated.ts:1]: We need to spend time on fixing this error and the copy/paste should only be the very last resort.

What happens if you put `"use client"` into the `usePagesOrInfinite` file?

File: packages/nextjs/src/shared/url.ts
Comment (SokratisVidros) [inline packages/nextjs/src/shared/url.ts:1]: ❓ What's the latest update on this? Is @clerk/shared still not tree-shakeable?

**Review by SokratisVidros — APPROVED** [review]

File: packages/nextjs/src/shared/deprecated.ts
Comment (dimkl) [reply packages/nextjs/src/shared/deprecated.ts:1]: I wanted to avoid introducing changes in `@clerk/shared` that would require extensive testing in this PR so I followed the naive approach of copy / paste to proceed with this PR objective and introduce another PR to support importing `@clerk/shared` in server-side frameworks.
Based on your comment, if we consider blocking this PR until the @clerk/shared is fixed and used I have opened the following PR in which we should also check if the @clerk/shared is tree-shakeable or not before we merge it: https://github.com/clerkinc/javascript/pull/1767
cc: @SokratisVidros

File: packages/nextjs/src/shared/url.ts
Comment (dimkl) [reply packages/nextjs/src/shared/url.ts:1]: I will investigate it as part of https://github.com/clerkinc/javascript/pull/1767

Comment (dimkl) [reply packages/nextjs/src/shared/url.ts]: It seems like `@clerk/shared` is not entirely tree-shakeable due to `window.__SWR_DEVTOOLS_USE__ = undefined;`
I used the `@next/bundle-analyzer` and ran our A`NALYZE=true npm run build` to generate the bundle analysis below (the screenshot is part of the client.html analysis).
Screenshot 2023-09-27 at 17 49 19

```
// next.config.js
const withBundleAnalyzer = require('@next/bundle-analyzer')({
  enabled: process.env.ANALYZE === 'true',
})
module.exports = withBundleAnalyzer(nextConfig) 
```

I've created a discussion in the team channel to resolve this.

ref: https://github.com/clerkinc/javascript/blob/0a17322842393670344369b0217ed3e989f912bb/packages/shared/src/hooks/clerk-swr.ts#L4C1-L4C1

File: packages/nextjs/src/shared/deprecated.ts
Comment (dimkl) [reply packages/nextjs/src/shared/deprecated.ts]: I have tested adding a deprecation in `authMiddleware` and `getAuth` and it seems that the tree-shaking works as expected and the above error is fixed!
I have also tried the `deprecated()` helper in `getServerSideProps` helper of a page and there is also no error. it seems that we can use it safely in server context.

File: packages/nextjs/src/shared/url.ts
Comment (dimkl) [reply packages/nextjs/src/shared/url.ts]: It seems that tree-shaking is fixed. 
The above dependency still exists in the client analysis because it is being used in some `@clerk/clerk-react` hooks that are being used by the clerk UI components.

**Comment by clerk-cookie** [comment]
This PR has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**
